maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/CheckDependencySnapshotsPhase.java

```diff
@@ -305,11 +305,14 @@ private static Artifact getArtifactFromMap( Artifact artifact, Map<String, Artif
     private static boolean checkArtifact( Artifact artifact, ReleaseDescriptor releaseDescriptor )
     {
         String versionlessKey = ArtifactUtils.versionlessKey( artifact.getGroupId(), artifact.getArtifactId() );
+        String releaseDescriptorResolvedVersion = releaseDescriptor.getDependencyReleaseVersion( versionlessKey );
         
         // We are only looking at dependencies external to the project - ignore anything found in the reactor as
         // it's version will be updated
         boolean result = artifact.isSnapshot()
-                && !artifact.getBaseVersion().equals( releaseDescriptor.getProjectOriginalVersion( versionlessKey ) );
+                && !artifact.getBaseVersion().equals( releaseDescriptor.getProjectOriginalVersion( versionlessKey ) )
+                        && ( releaseDescriptorResolvedVersion == null
```

Comment:
All the conditions make it hard to understand. How about a new if-statement

    if ( result ) {
      result = cmdlinepropertycheck
    }

I noticed the name `result` is kind o confusing as well. I'm thinking about something like `unallowedVersion`

Reply:
Hi Robert, I tried to simplify the if statement by introducing another local variable.